### PR TITLE
Switch to Go v1.14.4

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

Use Go version 1.14.4

Tested with 0.14.4 version of `golang/go` plugin

### CODE REVIEW

**Comment by benoitf** [comment]
Hi, why do we inherit from `FROM node:13.13.0-alpine` ?

Can't we inherit from `https://hub.docker.com/_/golang` images ?

**Comment by vitaliy-guliy** [comment]
> Hi, why do we inherit from `FROM node:13.13.0-alpine` ?
> 
> Can't we inherit from `https://hub.docker.com/_/golang` images ?

It's difficult to say.
Let's try whether it possible or not, with creating another PR if so.

**Comment by benoitf** [comment]
I think it's come to the time where we were depending on the nodejs for the theia endpoint, but now that it's a static binary we should drop it to reuse as much as possible official images.

**Comment by ericwill** [comment]
> > Hi, why do we inherit from `FROM node:13.13.0-alpine` ?
> > Can't we inherit from `https://hub.docker.com/_/golang` images ?
> 
> It's difficult to say.
> Let's try whether it possible or not, with creating another PR if so.

Let's try doing it now as part of this PR.

**Comment by vitaliy-guliy** [comment]
Reworked and tested with plugin `golang/go:0.14.4`

Devfile to test
```
---
apiVersion: 1.0.0
metadata:
  generateName: golang-
projects:
-
  name: example
  source:
    type: git
    location: https://github.com/golang/example.git
  clonePath: src/github.com/golang/example/
components:
-
  type: chePlugin
  # id: ms-vscode/go/latest
  reference: >-
    https://gist.githubusercontent.com/vitaliy-guliy/64a0233fe6ae9159cc8001253927c700/raw/3fc73c9ce51f907736866bf31fc8517031201da9/meta.yaml
  # reference: >-
  #   https://raw.githubusercontent.com/eclipse/che-plugin-registry/master/v3/plugins/golang/go/0.14.4/meta.yaml
  alias: go-plugin
  memoryLimit: 512Mi
  env:
    - value: 'off'
      name: GO111MODULE
  preferences:
    go.lintTool: 'golangci-lint'
    go.lintFlags: '--fast'
    go.useLanguageServer: true
-
  type: dockerimage
  # this version is used in the plugin
  image: quay.io/eclipse/che-golang-1.12:nightly
  alias: go-cli
  env:
    - name: GOPATH
      # replicate the GOPATH from the plugin
      value: /go:$(CHE_PROJECTS_ROOT)
    - name: GOCACHE
      # replicate the GOCACHE from the plugin, even though the cache is not shared
      # between the two
      value: /tmp/.cache
  endpoints:
    - name: '8080/tcp'
      port: 8080
  memoryLimit: 512Mi
  mountSources: true
commands:
-
  name: run outyet
  actions:
  - type: exec
    component: go-cli
    command: go get -d && go run main.go
    workdir: ${CHE_PROJECTS_ROOT}/src/github.com/golang/example/outyet
-
  name: stop outyet
  actions:
  - type: exec
    component: go-cli
    command: kill $(pidof go)
-
  name: test outyet
  actions:
  - type: exec
    component: go-cli
    command: go test
    workdir: ${CHE_PROJECTS_ROOT}/src/github.com/golang/example/outyet
-
  name: run current file
  actions:
  - type: exec
    component: go-cli
    command: go get -d && go run ${file}
    workdir: ${fileDirname}
-
  name: Debug current file
  actions:
  - type: vscode-launch
    referenceContent: |
      {
        "version": "0.2.0",
        "configurations": [
          {
            "name": "Debug current file",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${fileDirname}"
          }
        ]
      }

```

**Review by svor — APPROVED** [review]
Language server functionality works nice.
@vitaliy-guliy why in your devfile do you use `che-golang-1.12:nightly` not `che-golang-1.14`, it is not good if you use different versions of golang for sidecar container and dev container

File: Dockerfile [hunk 1]
```diff
@@ -70,22 +36,23 @@ RUN go get -u -v github.com/ramya-rao-a/go-outline && \
...
     GO111MODULE=on go get -v golang.org/x/tools/gopls@latest && \
     go build -o /go/bin/gocode-gomod github.com/stamblerre/gocode && \
     chmod -R 777 /go && \
-    apk del .build-deps && \
-    mkdir /.cache && chmod -R 777 /.cache && \
+    mkdir -p /.cache && chmod -R 777 /.cache && \
+    mkdir -p /usr/local/go && chmod -R 777 /usr/local/go && \
     cd /usr/local/go && wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.22.2
 
 ENV GOPATH /go
```
Comment (svor) [inline Dockerfile:50]: looks like this ENV was already set in base image https://github.com/docker-library/golang/blob/11eef5dfbc243e2c172c0ed0a38f65c0c756486d/1.14/stretch/Dockerfile#L75

**Comment by vitaliy-guliy** [comment]
> Language server functionality works nice.
> @vitaliy-guliy why in your devfile do you use `che-golang-1.12:nightly` not `che-golang-1.14`, it is not good if you use different versions of golang for sidecar container and dev container

Yep, indeed. Will change it in the devfile.

**Comment by vitaliy-guliy** [comment]
I don't know who adds `/projects` to the `GOPATH`.
That's what I got after removing the `ENV` from the Dockerfile 

```
GOPATH='/go:/projects'
```

Screenshot from 2020-07-09 18-20-32


I would like to keep it as it is (just `/go`).
@svor @ericwill  WDYT?

**Comment by benoitf** [comment]
@vitaliy-guliy introduced there: https://github.com/eclipse/che-devfile-registry/pull/3/files

**Comment by svor** [comment]
@vitaliy-guliy this env was set in devfile for dev container https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/go/devfile.yaml#L33 and in plugin's meta.yaml https://github.com/svor/che-plugin-registry/blob/sv/test3/v3/plugins/ms-vscode/go/0.11.4/meta.yaml#L19 for sidecar as well.
And it has correct value, I don't see any problems with `/go:/projects`.

**Comment by ericwill** [comment]
> And it has correct value, I don't see any problems with `/go:/projects`.

Agreed.

**Comment by vitaliy-guliy** [comment]
> @vitaliy-guliy introduced there: https://github.com/eclipse/che-devfile-registry/pull/3/files

I use custom devfile to test which is in the comment https://github.com/che-dockerfiles/che-sidecar-go/pull/18#issuecomment-655436869

**Pull request merged**